### ISSUE TITLE
False negative: attrs classes and hashability

### ISSUE THREAD

**Issue by Tinche** [python/mypy#16550]
Looks like non-frozen attrs classes are considered hashable by default. But they aren't.

The following snippet type-checks but explodes at runtime:

```python
from collections.abc import Hashable

from attrs import define


@define
class A:
    a: int


def my_func(c: Hashable) -> None:
    {c}


my_func(A(1))
```

(The behavior can be [customized](https://www.attrs.org/en/stable/hashing.html).)

This might be an easy one to fix. I can grab it when I have time, or it may be an easy first issue for someone?
